File: SpringBootFunctionalities/SpringSecurityPractice3/src/main/java/com/example/demo/entities/Fooditem.java
package com.example.demo.entities;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@ToString
@Entity
@Table(name="fooditems")
public class Fooditem {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private int food_id;
	
	@Column
	private String food_name;
	
	@Column
	private int qty;
	
	@Column
	private String unit;
	
	@Column 
	private int calories;
	
	@Column
	private int protein;
	
	@JsonIgnoreProperties
	@ManyToOne
	@JoinColumn(name="category_id")
	private Category category;
	
}

File: .NetFunctionalities/CaloryCoach/TrainerUseCases/Models/Fooditem.cs
using System;
using System.Collections.Generic;

namespace TrainerUseCases.Models
{
    public partial class Fooditem
    {
        public Fooditem()
        {
            Mealfooditemtransactions = new HashSet<Mealfooditemtransaction>();
        }

        public int FoodId { get; set; }
        public string? FoodName { get; set; }
        public int? Qty { get; set; }
        public string? Unit { get; set; }
        public int? Calories { get; set; }
        public int? Protein { get; set; }
        public int? CategoryId { get; set; }

        public virtual Category? Category { get; set; }
        public virtual ICollection<Mealfooditemtransaction> Mealfooditemtransactions { get; set; }
    }
}
